### ISSUE TITLE
[Request] Service IDs in CustomIdGenerator

### ISSUE THREAD

**Issue by aequasi** [doctrine/DoctrineBundle#759]
Allow passing of service ids to CustomIdGenerator annotation.

Would love to be able to do 

```php
/**
 * @ORM\CustomIdGenerator(service="my.id.generator")
 */
public $id;
```

Or something along those lines.

**Comment by oleg-andreyev** [comment]
It would be a great feature

**Comment by kimhemsoe** [comment]
I am closing this as i believe it does not belong here.

I have not done much investigation but i think to support this you would need to start in the ORM repo.
You can take a look at https://github.com/doctrine/doctrine2/blob/89e39b819281f2e537824b0851ca93f69a96826b/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php#L507

You need to convince the right people that instancing of identity generators should be configurable (Maybe it already is?). :D

Start by explaining your use case, suggest solutions and be ready to implement it yourself. 
A possible acceptable solution is simple and it does not introduce knowledge about other frameworks and etc into doctrine (symfony in this case). 

If/when you succeed, the next step is adding it either here or in the symfony doctrine bridge. I believe it belong in the bridge if it is. Alternative solution is implementing your own library/bundle.

**Comment by aequasi** [comment]
Was hoping it was something that could be done without editing the ORM as that could take years ;). 

Alrighty though, I'll look at asking them!

**Comment by nicolas-grekas** [comment]
Implemented in #1284
